src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -407,7 +408,13 @@ public function getQueueableClass()
      */
     public function getQueueableIds()
     {
-        return $this->modelKeys();
+        if ($this->isEmpty()) {
+            return [];
+        }
+
+        return $this->first() instanceof Pivot
+                    ? $this->map->getQueueableId()->all()
+                    : $this->modelKeys();
```

Comment:
What about adding `getQueueableId()` to the `Model` base class, and overriding it in `Pivot`? Then you can always map to `getQueueableId()` and get rid of the `instanceof`. :wink: